### PR TITLE
Priority ERROR units never die possibility

### PR DESCRIPTION
**Pull request by AkiraforARMA**
**When merged this pull request will:**
- Resolve bug when units with (no fatal) low limb damage in their last life become immortal.

Explain:
Cases when _lifesLeft goes below 0. 
Low or limb damage, then recovered with PAK from unconcious & revivestate.

Versions: **All versions, current master allows this bug**

Trace:
Ok I think I can trace what is happening.
Settings:
-preventinstadeath =False
-enablerevive = True
-amountOfReviveLives = 1

Situation:

_Unit has been revived once with PAK so _LifesLeft = 0
_ _Unit receives accumulation of limb damage.
Process:

New damage on limbs that is not fatal.
From: fnc_handleDamage.sqf goes to:
(fnc_handleDamage.sqf)Line 70 --> fnc_handleDamage_caching.sqf
(fnc_handleDamage.sqf)Line 109 --> fnc_vitalLoop.sqf
Stays on vital loop and:
From -- fnc_handleDamage_cahing.sqf goes to
(fnc_handleDamage_cahing.sqf)Line 92 --> fnc_handleDamage_advancedSetDamage.sqf then goes to:
(fnc_handleDamage_advancedSetDamage.sqf )Lines 28 to 31 --> fnc_setHitPointDamage.sqf then if there is damage accumulation:
(fnc_setHitPointDamage.sqf )Line 77 --> fnc_setDead.sqf With **NO** force parameter.
(fnc_setDead.sqf)Line 31 is OK,
(fnc_setDead.sqf)Line 32 NOT OK (Unit is not in revive state YET) then
(fnc_setDead.sqf)Lines 43 to 48 --> Unit enters reviveloop and go unconscious -fnc_reviveStateLoop.sqf
(fnc_reviveStateLoop.sqf)Line 45 - If PAK is used _ LifesLeft = -1
From there and now on conditions never match with line 35 of fnc_setDead.sqf so:
Unit never dies.

Please Review to confirm:
fnc_reviveStateLoop.sqf
fnc_setDead.sqf
fnc_handleDamage.sqf

Please tell me what more information is needed to add it.

Thanks.

### CODE REVIEW

File: addons/medical/functions/fnc_setDead.sqf [hunk 1]
```diff
@@ -32,7 +32,7 @@ if (((_reviveVal == 1 && {[_unit] call EFUNC(common,isPlayer)} || _reviveVal ==
     if (_unit getVariable [QGVAR(inReviveState), false]) exitwith {
         if (GVAR(amountOfReviveLives) > 0) then {
             _lifesLeft = _unit getVariable[QGVAR(amountOfReviveLives), GVAR(amountOfReviveLives)];
-            if (_lifesLeft == 0) then {
+            if (_lifesLeft <= 0) exitwith {
```
Comment (AkiraforARMA) [inline addons/medical/functions/fnc_setDead.sqf]: Just anti-error changes, further depuration is needed to find why is happening.

Comment (AkiraforARMA) [reply addons/medical/functions/fnc_setDead.sqf]: Ok I think I can trace what is happening.
Settings:
-preventinstadeath =False
-enablerevive = True
-amountOfReviveLives = 1

Situation:
- _Unit has been revived once with PAK so _LifesLeft = 0
_ _Unit receives accumulation off limb damage.

Process:
1) New damage on lims that is not fatal.
 From: fnc_handleDamage.sqf goes to:
       Line 70   -- fnc_handleDamage_caching.sqf
       Line 109 -- fnc_vitalLoop.sqf
Stays on vital loop and:
   From -- fnc_handleDamage_cahing.sqf goes to
       Line 92 -- fnc_handleDamage_advancedSetDamage.sqf then goes to:
           Lines 28 to 31 -- fnc_setHitPointDamage.sqf then if there is damage accumulation:
                Line 77 -- fnc_setDead.sqf With NO force parameter.
                      Line 31 is OK, Line 32 NOT OK (Unit is not in revive state YET) then
                      Lines 43 to 48 - Unit enters reviveloop and go unconscious -fnc_reviveStateLoop.sqf
                           Line 45 - If PAK is used _ LifesLeft = -1
From there and now on conditions never match with line 35 of fnc_setDead.sqf so:
Unit never dies.

Comment (AkiraforARMA) [reply addons/medical/functions/fnc_setDead.sqf]: Info added.

Comment (jonpas) [reply addons/medical/functions/fnc_setDead.sqf]: How does this `exitWith` change anything? There is nothing else in this block to execute after it.

**Pull request merged**